master/buildbot/www/resource.py

```diff
@@ -69,7 +69,7 @@ def writeErrorDefault(msg, errcode=400):
         if writeError is None:
             writeError = writeErrorDefault
         try:
-            d = _callable(request)
+            d = defer.maybeDeferred(_callable, request)
```

Comment:
seems like a good candidate for rewrite to `defer.inlineCallbacks` or coroutine. Not for this PR though.